packages/block-library/src/navigation/edit/index.js

```diff
@@ -848,48 +837,103 @@ function Navigation( {
 		<EntityProvider kind="postType" type="wp_navigation" id={ ref }>
 			<RecursionProvider uniqueId={ recursionId }>
 				<InspectorControls>
-					<PanelBody title={ __( 'Menu' ) }>
-						<NavigationMenuSelector
-							currentMenuId={ ref }
-							clientId={ clientId }
-							onSelectNavigationMenu={ ( menuId ) => {
-								handleUpdateMenu( menuId );
-							} }
-							onSelectClassicMenu={ async ( classicMenu ) => {
-								const navMenu = await convertClassicMenu(
-									classicMenu.id,
-									classicMenu.name,
-									'draft'
-								);
-								if ( navMenu ) {
-									handleUpdateMenu( navMenu.id, {
-										focusNavigationBlock: true,
-									} );
+					{ isOffCanvasNavigationEditorEnabled && (
+						<PanelBody>
+							<HStack className="wp-block-navigation-off-canvas-editor__header">
+								<Heading
+									className="wp-block-navigation-off-canvas-editor__title"
+									level={ 2 }
+								>
+									{ __( 'Menu' ) }
+								</Heading>
+								<NavigationMenuSelector
+									currentMenuId={ ref }
+									clientId={ clientId }
+									onSelectNavigationMenu={ ( menuId ) => {
+										handleUpdateMenu( menuId );
+									} }
+									onSelectClassicMenu={ async (
+										classicMenu
+									) => {
+										const navMenu =
+											await convertClassicMenu(
+												classicMenu.id,
+												classicMenu.name,
+												'draft'
+											);
+										if ( navMenu ) {
+											handleUpdateMenu( navMenu.id, {
+												focusNavigationBlock: true,
+											} );
+										}
+									} }
+									onCreateNew={
+										createUntitledEmptyNavigationMenu
+									}
+									createNavigationMenuIsSuccess={
+										createNavigationMenuIsSuccess
+									}
+									createNavigationMenuIsError={
+										createNavigationMenuIsError
+									}
+									/* translators: %s: The name of a menu. */
+									actionLabel={ __( "Switch to '%s'" ) }
+								/>
+							</HStack>
+
+							<OffCanvasEditor
+								blocks={ innerBlocks }
+								isExpanded={ true }
+								selectBlockInCanvas={ false }
+							/>
+						</PanelBody>
+					) }
+					{ ! isOffCanvasNavigationEditorEnabled && (
+						<PanelBody title={ __( 'Menu' ) }>
+							<NavigationMenuSelector
```

Comment:
It might be worth setting some of this stuff as a const, so that we don't need to duplicate it.

Reply:
I agree but we should do it in a separate refactoring PR targeting trunk. Here we're only exploring this UX pattern.